src/toxcore/messenger/packet/typing.rs

```diff
@@ -0,0 +1,75 @@
+/*! Typing struct.
+*/
+
+use nom::le_u8;
+
+use crate::toxcore::binary_io::*;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+/// Typing status of user
+pub enum TypingStatus {
+    /// Not typing
+    NotTyping = 0,
+    /// Typing
+    Typing,
+}
+
+impl FromBytes for TypingStatus {
+    named!(from_bytes<TypingStatus>,
+        switch!(le_u8,
+            0 => value!(TypingStatus::NotTyping) |
+            1 => value!(TypingStatus::Typing)
+        )
+    );
+}
+
+/** Typing is a struct that holds typing status of user.
+
+This packet is used to transmit sender's typing status to a friend.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x33`
+`1`       | Typing status(0 = not typing, 1 = typing)
+
+*/
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Typing {
+    status: TypingStatus,
```

Comment:
How about unifying with https://github.com/tox-rs/tox/pull/316/files#diff-50f8bfc82d815d2951af7e519042f913R29 - you used short form of struct there, but full here.

Reply:
done

Reply:
And the same question for UserStatus then :)